docs/flow/reference/components/discovery.kubernetes.md

```diff
@@ -0,0 +1,163 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/discovery.kubernetes
+title: discovery.kubernetes
+---
+
+# discovery.kubernetes
+
+`discovery.kubernetes` allows you to find scrape targets from kubernetes resources. It watches cluster state, and ensures targets are continually synced with what is currently running in your cluster.
+
+Internally, this component uses kubernetes service discovery from prometheus. [The documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) on their site offers further details on the metadata labels applied to targets and other usage considerations.
```

Comment:
```suggestion
Internally, this component uses Kubernetes Service Discovery from Prometheus. [The documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) on their site offers further details on the metadata labels applied to targets and other usage considerations.
```

I'd prefer avoiding linking out to Prometheus documentation to avoid issues where we become out of sync if upstream updates before we have a chance to support something. Let's reproduce what we need to explain to users here instead of linking out.